database/joda_time.ksy

```diff
@@ -0,0 +1,177 @@
+meta:
+  id: joda_time
+  title: Joda-Time timezone
+  application:
+    - Joda-Time library
+  license: Apache-2.0
+  ks-version: 0.9
+  encoding: utf-8
+  endian: be
+
+doc: Joda-Time is a time manipulation library for Java. It stores info about timezones in own binary format. The timezones are embedded into the jars (can be downloaded from https://github.com/JodaOrg/joda-time/releases/) and can be unpacked from them.
+doc-ref: https://github.com/JodaOrg/joda-time/blob/e37246fb936b2c242812235dff53c749e18ec200/src/main/java/org/joda/time/tz/DateTimeZoneBuilder.java#L110
+seq:
+  - id: encoding
+    type: str
+    size: 1
+  - id: payload
+    type:
+      switch-on: encoding
+      cases:
+        '"F"': fixed
+        '"C"': precalculated  # cached
+        '"P"': precalculated
+types:
+  pas_str:
+    doc-ref: https://docs.oracle.com/javase/7/docs/api/java/io/DataInput.html#readUTF()
+    seq:
+      - id: length
+        type: u2
+      - id: value
+        type: str
+        size: length
+  fixed:
+    seq:
+      - id: name_key
+        -orig-id: nameKey
+        type: pas_str
+      - id: wall_offset
+        -orig-id: wallOffset
+        type: millis
+      - id: standard_offset
+        -orig-id: standardOffset
+        type: millis
+  precalculated:
+    -orig-id: PrecalculatedZone
+    seq:
+      - id: pool_size
+        -orig-id: poolSize
+        type: u2
+      - id: pool
+        type: pas_str
+        repeat: expr
+        repeat-expr: pool_size
+      - id: size
+        -orig-id: size
+        type: u4
+      - id: smth
+        type: smth
+        repeat: expr
+        repeat-expr: size
+      - id: tail_present
+        type: u1
+      - id: tail
+        type: dst_zone
+        if: tail_present != 0
+
+    types:
+      smth:
+        seq:
+          - id: transition
+            type: millis
+          - id: wall_offsets
+            type: millis
+          - id: standard_offsets
+            type: millis
+          - id: index
+            type:
+              switch-on: _parent.pool_size < 256
+              cases:
+                true: u1
+                false: u2
+
+  millis:
+    seq:
+      - id: v
+        type: u1
+      - id: payload
+        type:
+          switch-on: type
+          cases:
+            0: form_0(v)
+            1: form_1(v)
+            2: form_2(v)
+            3: form_3
+    instances:
+      type:
+        value: v >> 6
+
+    types:
+      form_0:
+        params:
+          - id: v
+            type: u1
+        instances:
+          value:
+            value: ((v << (32 - 6)) >> (32 - 6)) * (30 * 60000)
```

Comment:
This method of sign extension is poorly portable and generally isn't guaranteed to work in target languages. There is also no guarantee that the default integer type will remain `s4` in the future.  I recommend using the approach I used in `vlq_base128_le` instead:

https://github.com/kaitai-io/kaitai_struct_formats/blob/4c4bfc0709a8a2f6d771a6dc3094ccfa17def3b3/common/vlq_base128_le.ksy#L61-L65

However, you need to take only the lowest 6 bits from `v`. If the highest 2 bits of the `v` byte would be set, it would not work correctly anymore (there are no longer any _back and forth_ shifts which would trim it from both sides).

Reply:
Yeah, I know; it was just verbatim copy from the source of Joda Time. I gonna refactor it.